src/api/routes/block.ts

```diff
@@ -21,18 +22,18 @@ export function createBlockRouter(db: DataStore): RouterWithAsync {
     const limit = parseBlockQueryLimit(req.query.limit ?? 20);
     const offset = parsePagingQueryInput(req.query.offset ?? 0);
 
-    const blocks = await db.getBlocks({ offset, limit });
+    const { results: blocks, total } = await db.getBlocks({ offset, limit });
     // TODO: fix duplicate pg queries
-    const result = await Bluebird.mapSeries(blocks.results, async block => {
+    const results = await Bluebird.mapSeries(blocks, async block => {
       const blockQuery = await getBlockFromDataStore(block.block_hash, db);
       if (!blockQuery.found) {
         throw new Error('unexpected block not found -- fix block enumeration query');
       }
       return blockQuery.result;
     });
-
+    const response: BlockResults = { limit, offset, total, results };
```

Comment:
This could also implement the `validate` fn

Reply:
I'll leave this as a TODO for now, hopefully tackle later with a more integrated `openapi <-> express` solution